### PR TITLE
Add some HardwareOptions #2

### PR DESCRIPTION
**Pull request by Speed1**
Add some HardwareOptions

### CODE REVIEW

File: snipeit.go [hunk 1]
```diff
@@ -216,8 +216,11 @@ func (c *Client) Location(id int64) (*Location, *http.Response, error) {
 // HardwareOptions specifies a subset of optional query parameters
 // for listing assets.
 type HardwareOptions struct {
-	LocationID int `url:"location_id,omitempty"`
-	CompanyID  int `url:"company_id,omitempty"`
+	Limit      int    `url:"limit,omitempty"`
+	Search     string `url:"search,omitempty"`
+	CategoryID int    `url:"categoryid,omitempty"`
```
Comment (palmamartin) [inline snipeit.go]: This should be`url:"category_id,omitempty"`

File: snipeit.go [hunk 2]
```diff
@@ -216,8 +216,11 @@ func (c *Client) Location(id int64) (*Location, *http.Response, error) {
...
+	LocationID int    `url:"location_id,omitempty"`
+	CompanyID  int    `url:"company_id,omitempty"`
```
Comment (palmamartin) [inline snipeit.go]: I think at this point we should add all supported parameters for the hardware endpoint: Missing `offset`, `order_number`, `sort`, `order`, `model_id`, `manufacturer_id`, `status` and `status_id`

**Comment by Speed1** [comment]
I have added now all hardware and location options ;)

File: snipeit.go [hunk 3]
```diff
@@ -216,8 +219,19 @@ func (c *Client) Location(id int64) (*Location, *http.Response, error) {
 // HardwareOptions specifies a subset of optional query parameters
 // for listing assets.
 type HardwareOptions struct {
-	LocationID int `url:"location_id,omitempty"`
-	CompanyID  int `url:"company_id,omitempty"`
+	Limit          int    `url:"limit,omitempty"`
+	Offset         int    `url:"offset,omitempty"`
+	Search         string `url:"search,omitempty"`
+	Ordernumber    string `url:"order_number,omitempty"`
```
Comment (palmamartin) [inline snipeit.go]: Use camelcase for variable names composed from two or more words. e.g. `OrderNumber`

**Comment by palmamartin** [comment]
On last thing: Could you squash the 4 commits into a single one and name it something like: Adding Location and HardwareOptions...

**Comment by palmamartin** [comment]
LGMT!

**Pull request merged**